cypress/e2e/shared/dashboardsIndex.test.ts

```diff
@@ -252,6 +253,183 @@ describe('Dashboards', () => {
+                                      cy.visit(`${orgs}/${id}/dashboards-list`)
+                                      return cy.getByTestID('tree-nav')
+                                    })
+                                )
+                              })
+                          )
+                      )
+                  )
+                )
+              })
+          )
+        )
+      )
+    })
+
+    it('can clone a dashboard', () => {
+      cy.getByTestID('dashboard-card').should('have.length', 1)
+
+      // get graph in original view
+
+      cy.getByTestID('dashboard-card')
+        .first()
+        .within(() => {
+          cy.getByTestID(`label--pill ${labelName}`).should('be.visible')
+          cy.getByTestID('dashboard-card--name').click()
+        })
+
+      const viewGraphOrig = makeGraphSnapshot()
+
+      cy.getByTestID('nav-item-dashboards').click()
+
+      cy.getByTestID('dashboard-card')
+        .first()
+        .within(() => {
+          cy.getByTestID('context-menu-dashboard').click()
+        })
+
+      cy.getByTestID('context-clone-dashboard').click()
+
+      // Verify cloned dashboard contents
+
+      cy.getByTestID('page-title').should(
+        'contain.text',
+        `${localDashName} (clone 1)`
+      )
+      cy.getByTestID('variable-dropdown--Power').should('be.visible')
+      cy.getByTestID('variable-dropdown-input-typeAhead--Power').should(
+        'have.value',
+        'base'
+      )
+
+      makeGraphSnapshot().shouldBeSameAs(viewGraphOrig)
+
+      cy.getByTestID('cell--view-empty markdown').should(
+        'contain.text',
+        'The cat went here and there'
+      )
+
+      // Prep verification that changes in clone are not reflected in original,
+      // i.e. that they are fully decoupled
+      // change variable N.B. final assert is waiting on fix for #3287 see below
+      cy.getByTestID('variable-dropdown-input-typeAhead--Power').click()
+      cy.getByTestID('variable-dropdown--item')
+        .eq(4)
+        .click()
+      // const viewGraphCopy = makeGraphSnapshot()
+      cy.getByTestID('cell Name this Cell').within(() => {
+        cy.getByTestID('cell-context--toggle').click()
+      })
+
+      const replaceText =
+        '### William Blake\n#### Augeries of Innocense\n\n To see a World in a Grain of Sand\n\nAnd a Heaven in a Wild Flower\n\n...`'
+      cy.getByTestID('cell-context--note').click()
+      cy.getByTestID('markdown-editor')
+        .click()
+        .type('{ctrl}a')
+        .type(replaceText)
+
+      cy.getByTestID('save-note--button').click()
+
+      cy.getByTestID('nav-item-dashboards').click()
+
+      // Verify labels
+      cy.getByTestID('dashboard-card')
+        .eq(1)
+        .within(() => {
+          cy.getByTestID(`label--pill ${labelName}`).should('be.visible')
+          cy.getByTestID('dashboard-card--name').should(
+            'contain.text',
+            `${localDashName} (clone 1)`
```

Comment:
i don't know all the intricacies of what's on the screen at this point in the test, but it might be more straightforward to do something like:

```ts
cy.contains(`${localDashName} (clone 1)`)
```

or

```ts
cy.getByTestID('dashboard-card--name').contains(
  `${localDashName} (clone 1)`
)
```


[this SO answer provides some context](https://stackoverflow.com/a/50049358/92446)

Reply:
replaced with contains

Reply:
On second though I've decided to restore the `.should()` method.  `.should()` is used when making assertions, while `.contains()` is used more like a selector in a chain.  At this point in the test we just want to assert that the card was cloned with the correct name.